### PR TITLE
chore(scanner): limit Central and Matcher ingress to CentralServices

### PR DESCRIPTION
**Pull request by RTann**
## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

### Here I tell how I validated my change

CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml [hunk 1]
```diff
@@ -229,5 +229,25 @@ tests:
...
-    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.repository_to_cpe_url | assertThat(contains("https://central.stackrox.svc/api/extensions/scannerdefinitions"))
-    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.name_to_repos_url | assertThat(contains("https://central.stackrox.svc/api/extensions/scannerdefinitions"))
+    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.repository_to_cpe_url | assertThat(. == "https://central.stackrox.svc/api/extensions/scannerdefinitions?file=repo2cpe")
+    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.name_to_repos_url | assertThat(. == "https://central.stackrox.svc/api/extensions/scannerdefinitions?file=name2cpe")
+
+- name: "indexer should allow ingress from central and matcher when deployed as a central-service"
+  set:
+    scannerV4.disable: false
+  expect: |
+    .networkpolicys["scanner-v4-indexer"].spec.ingress | assertThat(length == 3)
```
Comment (RTann) [inline pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml:239]: how much work would it be to convert `networkpolicys` to `networkpolicies`?

File: image/templates/helm/shared/templates/02-scanner-v4-05-indexer-network-policy.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ spec:
     matchLabels:
       app: scanner-v4-indexer
   ingress:
+  {{- if ._rox.env.centralServices }}
```
Comment (mclasmeier) [inline image/templates/helm/shared/templates/02-scanner-v4-05-indexer-network-policy.yaml:17]: Just to make sure I get this right: indexer on central side shall be accessible from central and sensor while indexer when deployed on a secured cluster shall only be accessible from sensor.

File: pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml (hunk 1 above)
Comment (mclasmeier) [reply pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml:249]: Don't know from the top of my head. Would need to check in helmtest.

File: pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml [hunk 3]
```diff
@@ -229,5 +239,25 @@ tests:
   set:
     scannerV4.disable: false
   expect: |
-    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.repository_to_cpe_url | assertThat(contains("https://central.stackrox.svc/api/extensions/scannerdefinitions"))
-    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.name_to_repos_url | assertThat(contains("https://central.stackrox.svc/api/extensions/scannerdefinitions"))
+    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.repository_to_cpe_url | assertThat(. == "https://central.stackrox.svc/api/extensions/scannerdefinitions?file=repo2cpe")
+    .configmaps["scanner-v4-indexer-config"].data.["config.yaml"] | fromyaml | .indexer.name_to_repos_url | assertThat(. == "https://central.stackrox.svc/api/extensions/scannerdefinitions?file=name2cpe")
+
+- name: "indexer should allow ingress from central and matcher when deployed as a central-service"
```
Comment (mclasmeier) [inline pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml:245]: + sensor?

File: pkg/helm/charts/tests/securedclusterservices/testdata/helmtest/scanner-v4.test.yaml [hunk 4]
```diff
@@ -35,6 +35,25 @@ tests:
     .deployments["scanner-v4-indexer"] | assertThat(. == null)
     .deployments["scanner-v4-db"] | assertThat(. == null)
 
+- name: "scanner V4 is enabled should should only deploy indexer and db"
```
Comment (mclasmeier) [inline pkg/helm/charts/tests/securedclusterservices/testdata/helmtest/scanner-v4.test.yaml:38]: wording

**Review by dcaravel — APPROVED** [review]
LGTM

File: image/templates/helm/shared/templates/02-scanner-v4-05-indexer-network-policy.yaml (hunk 2 above)
Comment (RTann) [reply image/templates/helm/shared/templates/02-scanner-v4-05-indexer-network-policy.yaml:17]: Correct. Note: we rely on Sensor -> Indexer for delegated scanning and OpenShift Internal Registry scanning purposes, so we need to always allow this.

**Pull request merged**